### DISCUSSION TITLE
what is this "rid" is the url of app?  ?a=normal&get=f_1636&rid=many_numbers&order=DESC&state=3

### DISCUSSION THREAD

**Discussion by rezad1393** [FreshRSS/FreshRSS#6579] [Q&A] [answered]
what is this "rid" is the url of app?  ?a=normal&get=f_1636&rid=many_numbers&order=DESC&state=3

**Comment by Frenzie** [comment]
Request ID, see https://github.com/FreshRSS/FreshRSS/pull/3208 for background info.

**Reply by rezad1393** [reply]
> Request ID, see #3208 for background info.

I have no idea what that is (not a dev) but it is needed?

**Reply by Frenzie** [reply, marked as answer]
It's necessary to prevent [this](https://github.com/FreshRSS/FreshRSS/pull/3096#issuecomment-654731822) situation:

> If a user opens FRSS in two tabs and sends two requests in parallel, we could have a situation where:
> 
>     * request A sets a notification
> 
>     * request B sets a notification
> 
>     * request A returns and shows the notification set by request B
> 
>     * request B returns and shows nothing
> 
> 
> It's maybe not a big deal, but by removing the lock mechanism, we introduce a risk of race conditions in the code.

(although wouldn't you still have that situation by duplicating the tab or something…? but that aside)
